core/templates/services/user-backend-api.service.spec.ts

```diff
@@ -0,0 +1,294 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests that the user service is working as expected.
+ */
+
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { CsrfTokenService } from 'services/csrf-token.service';
+
+import { UserBackendApiService } from 'services/user-backend-api.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { UserInfoObjectFactory } from 'domain/user/UserInfoObjectFactory';
+import { UrlService } from './contextual/url.service';
+
+class MockWindowRef {
+  _window = {
+    location: {
+      pathname: 'home'
+    }
+  };
+  get nativeWindow() {
+    return this._window;
+  }
+}
+
+describe('User Backend Api Service', () => {
+  let userBackendApiService: UserBackendApiService = null;
+  let urlInterpolationService: UrlInterpolationService = null;
+  let userInfoObjectFactory: UserInfoObjectFactory = null;
+  let urlService: UrlService = null;
+  let httpTestingController: HttpTestingController = null;
+  let csrfService: CsrfTokenService = null;
+  let windowRef: MockWindowRef = null;
+
+  beforeEach(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [{ provide: WindowRef, useValue: windowRef }]
+    });
+    httpTestingController = TestBed.get(HttpTestingController);
+    userBackendApiService = TestBed.get(UserBackendApiService);
+    urlInterpolationService = TestBed.get(UrlInterpolationService);
+    userInfoObjectFactory = TestBed.get(UserInfoObjectFactory);
+    urlService = TestBed.get(UrlService);
+    csrfService = TestBed.get(CsrfTokenService);
+
+    spyOn(csrfService, 'getTokenAsync').and.callFake(
+      () =>{
+        return Promise.resolve('sample-csrf-token');
+      });
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should return userInfo data', fakeAsync(() => {
+    // Creating a test user for checking profile picture of user.
+    var sampleUserInfoBackendObject = {
```

Comment:
Can you use `const` or `let` depending on the use of this variable. Here and everywhere else (where var is `used`) in the migrated code.